### PR TITLE
Test data integrity

### PR DESCRIPTION
**Pull request by nitisht**
- Send logs to Parseable
- Wait for sync
- Download parquet from the store created by Parseable for the minute
- Compare the sent logs with the ones loaded from the downloaded parquet

### CODE REVIEW

**Comment by nitisht** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by nitisht** [comment]
cc @nikhil-bijapur can you sign the CLA with the comment

**Comment by theteachr** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by nitisht** [comment]
recheck

**Comment by theteachr** [comment]
- [ ] Mark test failure after `slog.Error` calls

File: integrity_test.go [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+	downloadedFileNames := make([]string, 0, 10)
+
+	for objectInfo := range client.ListObjectsV2(config.Bucket, stream, true, nil) {
+		key := objectInfo.Key
+
+		if !isParquetFile(key) {
+			continue
+		}
+
+		parquetObject, err := client.GetObject("integrity-test", key, minio.GetObjectOptions{})
```
Comment (nitisht) [inline integrity_test.go]: Dont hard code the bucket name here

File: integrity_test.go [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+	rowCount := len(actualFlogs)
+
+	for i, expectedFlog := range flogs {
+		// The rows in parquet written by Parseable will be latest first, so we
+		// compare the first of ours with the last of what we got from Parseable's
+		// store.
+		actualFlog := actualFlogs[rowCount-i-1].Deref()
+		require.Equal(t, actualFlog, expectedFlog)
+	}
```
Comment (trueleo) [inline integrity_test.go:140]: I think here the stream can safely deleted after the test has finished. Otherwise LGTM :+1:

**Comment by nitisht** [comment]
@trueleo have you tested? does it work as expected

**Comment by trueleo** [comment]
@theteachr It seems to be failing with 

```
docker run --network=host trueleo8/quest:latest smoke http://localhost:8000 admin admin 
=== RUN   TestIntegrity
--- FAIL: TestIntegrity (120.11s)
2023/12/19 12:21:54 ERROR couldn't create MinIO client error="address http://localhost:9000: too many colons in address"
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7b5822]

goroutine 19 [running]:
testing.tRunner.func1.2({0x8025a0, 0xbac4d0})
        /usr/local/go/src/testing/testing.go:1545 +0x238
testing.tRunner.func1()
        /usr/local/go/src/testing/testing.go:1548 +0x397
panic({0x8025a0?, 0xbac4d0?})
        /usr/local/go/src/runtime/panic.go:914 +0x21f
quest.downloadParquetFiles({0x7ffdf254beab, 0xa}, {{0x7ffdf254bed9, 0x15}, {0x7ffdf254befb, 0xa}, {0x7ffdf254bf12, 0xa}, {0x7ffdf254bf2b, 0x9}})
        /tests/integrity_test.go:155 +0x102
quest.TestIntegrity(0x0?)
        /tests/integrity_test.go:121 +0x47b
testing.tRunner(0xc000107380, 0x8c8670)
        /usr/local/go/src/testing/testing.go:1595 +0xff
created by testing.(*T).Run in goroutine 1
        /usr/local/go/src/testing/testing.go:1648 +0x3ad
```

I am running with default params and created a bucket named `parseable` . There is some data in that bucket now, so it seems ingest step worked but afterward that it failed on the next step.

File: main.sh [hunk 3]
```diff
@@ -26,11 +26,19 @@ vus=$6
 : "${vus:=10}"
 duration=$7
 : "${duration:="5m"}"
+minio_url=$8
+: "${minio_url:="http://localhost:9000"}"
```
Comment (theteachr) [inline main.sh]:
```suggestion
: "${minio_url:="localhost:9000"}"
```
@trueleo This should fix https://github.com/parseablehq/quest/pull/47#issuecomment-1862669882.

**Pull request merged**